### PR TITLE
New manpages.

### PR DESCRIPTION
**Pull request by sarahjelinek**
This change is [Reviewable](https://reviewable.io/reviews/marcinslusarz/pmemfile/27)

### CODE REVIEW

**Comment by marcinslusarz** [comment]
Review status: 0 of 5 files reviewed at latest revision, 12 unresolved discussions.

---

*[doc/libpmemfile-core.3.md, line 6 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlOR_1wMVoTE2Pxly:-KerlOR_1wMVoTE2Pxlz:bksp4vt) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L6)):*
> ```Markdown
> title: "libpmemfile-posix(3)"
> header: PMEM Library
> date: "pmemfile-posix API version 1.0.0
> ```

Not really a PMEM library :) Also API is not at version 1.0.

---

*[doc/libpmemfile-core.3.md, line 22 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlSGSgc85b6lqJmBp:-KerlSGSgc85b6lqJmBq:b-1usvsj) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L22)):*
> ```Markdown
> [comment]: <> (      from this software without specific prior written permission.)
> 
> [comment]: <> (THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HO<br />LDERS AND CONTRIBUTORS)
> ```

That shouldn't be here.

---

*[doc/libpmemfile-core.3.md, line 49 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlbE1klkMSsIMQ3Vd:-KerlbE233IoSj6Q1TkE:b3eh8ps) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L49)):*
> ```Markdown
> 
> # NAME #
> **libpmemfile-posix**-- transactional user space persistent memory file system
> ```

The library being transactional is an implementation detail. We don't expose it to the users of this library.

---

*[doc/libpmemfile-core.3.md, line 53 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerllGz_SZ9fHK7-WhJ:-KerllGz_SZ9fHK7-WhK:b32vk7) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L53)):*
> ```Markdown
> # SYNOPSIS #
> ```c
> #include libpmemfile-posix.h
> ```

<>

---

*[doc/libpmemfile-core.3.md, line 58 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KermCzdSuPDH4KzI2OB:-KermCzdSuPDH4KzI2OC:b-skzfov) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L58)):*
> ```Markdown
> 
> # DESCRIPTION #
> **libpmemfile-posix** provides a transactional file system for persistent memory(pmem) which runs in user-space. A persistent memory file system provides the ability to move data from DRAM to persistent memory.
> ```

s/transactional//
From DRAM?

---

*[doc/libpmemfile.1.md, line 44 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerrqxfB9SoTOrhOtgH:-KerrqxfB9SoTOrhOtgI:bfycr1e) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L44)):*
> ```Markdown
> 
> # NAME #
> **Pmemfile** - transactional persistent memory file system.
> ```

s/transactional//

---

*[doc/libpmemfile.1.md, line 102 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KersB-AZzPI0Vofic4_:-KersB-AZzPI0Vofic4a:b7sthab) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L102)):*
> ```Markdown
> 
> CLONE_VM
> 	Is Supported. However, even if a thread is created with shared
> ```

s/Supported/supported/

---

*[doc/libpmemfile.1.md, line 174 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KersKAjLXfTUqAfngI9:-KersKAjLXfTUqAfngIA:b-gqikr2) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L174)):*
> ```Markdown
> int  FD_ISSET(int fd, fd_set *set);
> void FD_SET(int fd, fd_set *set);
> void FD_ZERO(fd_set *set);
> ```

FD_*  are just macros and there's no way to fail them, so I think they should not be listed here.

---

*[doc/libpmemfile.1.md, line 261 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-Kersn-uW_-Nt1RjkXMt:-Kersn-vJpHKQ3Lcy35D:b-dg11mc) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L261)):*
> ```Markdown
> 
> # SPECIAL FILES #
> The system calls that manage block or character special files are not supported.
> ```

... when pathname points to pmemfile-backed filesystem.

---

*[doc/libpmemfile.1.md, line 323 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertPDoQdzY2XlG14Q4:-KertPDoQdzY2XlG14Q5:bxfy19t) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L323)):*
> ```Markdown
> 
> O_NOATIME
> 	Supported.
> ```

Should it be here if it's supported and the behavior is not modified?

---

*[doc/libpmemfile.1.md, line 490 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertnNWpfTDuJgX529m:-KertnNWpfTDuJgX529n:b-eu27xz) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L490)):*
> ```Markdown
> # RENAME #
> 
> All renameX() functions are support except:
> ```

supported

---

*[doc/libpmemfile.1.md, line 494 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertwBgWYlgnqU-aq6i:-KertwBhC3a5P03cQWaj:bqh5zko) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L494)):*
> ```Markdown
> ```c
> int renameat2(int olddirfd, const char *oldpath,
> 	      int newdirfd, const char *newpath, unsigned int flags);
> ```

Why renameat2 is not supported?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 6 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlOR_1wMVoTE2Pxly:-Ket8hAl4RQpS57Fn8oY:b-f6sii3) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L6)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Not really a PMEM library :) Also API is not at version 1.0.
</blockquote></details>

It will be a pmem library though , correct? Not NVML library. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 22 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlSGSgc85b6lqJmBp:-Ket8xn7bitlUNK6KOaz:bt2vr1b) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L22)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

That shouldn't be here.
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 44 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerrqxfB9SoTOrhOtgH:-KetIi9GViHnaQpa4Epd:b2oomly) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L44)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

s/transactional//
</blockquote></details>

Removed transactional. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 58 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KermCzdSuPDH4KzI2OB:-KetHzQxuvfhnenBr_ul:b-jcufjz) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L58)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

s/transactional//
From DRAM?
</blockquote></details>

I removed transactional. It would really be to and from DRAM. Is it the use of DRAM terms that's bothering you? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 53 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerllGz_SZ9fHK7-WhJ:-KetJACIObOE5o-gYeor:b-qssebl) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L53)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

<>
</blockquote></details>

fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 102 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KersB-AZzPI0Vofic4_:-KetJYAEJBAa23ZqCv8m:b-qssebl) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L102)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

s/Supported/supported/
</blockquote></details>

fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 261 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-Kersn-uW_-Nt1RjkXMt:-KetJtTUUa9mkRPIXVvD:b-t948zx) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L261)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

... when pathname points to pmemfile-backed filesystem.
</blockquote></details>

added.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by marcinslusarz** [comment]
*[doc/libpmemfile-core.3.md, line 58 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KermCzdSuPDH4KzI2OB:-KetJAhZrr7sp6Iq_RQz:b-i9wli8) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L58)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

I removed transactional. It would really be to and from DRAM. Is it the use of DRAM terms that's bothering you? 
</blockquote></details>

Well, when you are starting to use pmemfile-posix it means you are moving data from kernel file system to persistent memory file system, not from DRAM to pmem...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 323 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertPDoQdzY2XlG14Q4:-KetK5yvVw9Md_lPRR7R:bk6es75) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L323)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Should it be here if it's supported and the behavior is not modified?
</blockquote></details>

no :-). I will remove.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 494 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertwBgWYlgnqU-aq6i:-KetKJtkVY-JW3Zj7PyY:b-ndvhdu) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L494)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Why renameat2 is not supported?
</blockquote></details>

the only way we support it is if the flags value is 0, which makes it essentially rename().  So, we can either not support it or simply say we support it but none of the flags are supported. Which do you prefer?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 174 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KersKAjLXfTUqAfngI9:-KetLvNdimdAdlYUfQF-:bk83f8h) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L174)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

FD_*  are just macros and there's no way to fail them, so I think they should not be listed here.
</blockquote></details>

ok. removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by marcinslusarz** [comment]
Review status: 0 of 5 files reviewed at latest revision, 12 unresolved discussions.

---

*[doc/libpmemfile-core.3.md, line 6 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlOR_1wMVoTE2Pxly:-KetKnik-ZTt07r1pOsa:bssx49g) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L6)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

It will be a pmem library though , correct? Not NVML library. 
</blockquote></details>

Ok, I can't find any documentation about the meaning of this field and I don't see it used by pandoc, so it can be whatever value we want...

---

*[doc/libpmemfile.1.md, line 494 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertwBgWYlgnqU-aq6i:-KetN2ECqS6_qFg3aivp:b30rczx) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L494)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

the only way we support it is if the flags value is 0, which makes it essentially rename().  So, we can either not support it or simply say we support it but none of the flags are supported. Which do you prefer?
</blockquote></details>

We can easily support RENAME_NOREPLACE flag and RENAME_EXCHANGE is not that hard...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 49 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlbE1klkMSsIMQ3Vd:-KetHaia8qxuubwP0KW3:b-awesa3) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L49)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

The library being transactional is an implementation detail. We don't expose it to the users of this library.
</blockquote></details>

Ok, fair enough. I will remove it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 490 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertnNWpfTDuJgX529m:-KetSjkQOXHsRXuRxhNO:b-qssebl) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L490)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

supported
</blockquote></details>

fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 494 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertwBgWYlgnqU-aq6i:-KetSlfgeDWlg9czqe14:b-hf9419) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L494)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

We can easily support RENAME_NOREPLACE flag and RENAME_EXCHANGE is not that hard...
</blockquote></details>

If we want to support those flags then I will add it. Seems reasonable.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

File: doc/libpmemfile.1.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
 layout: manual
 Content-Style: 'text/css'
-title: libpmemfile(1)
-header: PMEMFILE
-date: pmemfile API version 0.1.0
+title: pmemfile(1)
+header: PMEMFile
+date: pmemfile API version 1.0.0
```
Comment (GBuella) [inline doc/libpmemfile.1.md]: So we would already call it 1.0 ?
I'm not quite confident about that.
How about waiting till most of the interfaces described in this manual work already  - e.g. pmemfile_truncate functioning?

**Comment by marcinslusarz** [comment]
I don't see any of the "fixed" changes. Forgot to push?

**Comment by sarahjelinek** [comment]
Haven't pushed the changes yet. I should have said 'will fix'. I will push a new set of man pages today.

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 58 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KermCzdSuPDH4KzI2OB:-Kf84Eno4ZdMkj1saX0V:b-p9uuoq) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L58)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Well, when you are starting to use pmemfile-posix it means you are moving data from kernel file system to persistent memory file system, not from DRAM to pmem...
</blockquote></details>

Actually, we are moving data to and from DRAM.  Even if we read from the kernel fs, it's read into a dram buffer, than we write that data to pmemfile, we write from a dram buffer.  When we use NVML we have in place persistence. With pmemfile we have a buffer in between. Which is what I am trying to say.  Maybe I will just pull this out since it's confusing. And, honestly, isn't important to a user. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 6 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-Kepxa-GINwDQeiZWb7i-r1-6:-Kf85lLtUt_r4i11mFRX:b-3w8qpu) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L6)):*
<details><summary><i>Previously, GBuella (Gábor Buella) wrote…</i></summary><blockquote>

So we would already call it 1.0 ?
I'm not quite confident about that.
How about waiting till most of the interfaces described in this manual work already  - e.g. pmemfile_truncate functioning?
</blockquote></details>

I changed it to 0.1.1

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile.1.md, line 494 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KertwBgWYlgnqU-aq6i:-Kf863PKm-2mJPbER-_6:bn7ihjf) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile.1.md#L494)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

If we want to support those flags then I will add it. Seems reasonable.
</blockquote></details>

Added that we support renameat2() with exceptions noted.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by sarahjelinek** [comment]
*[doc/libpmemfile-core.3.md, line 6 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KerlOR_1wMVoTE2Pxly:-Kf86GMMT5DDvX9ozRLN:b-gnskbs) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L6)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Ok, I can't find any documentation about the meaning of this field and I don't see it used by pandoc, so it can be whatever value we want...
</blockquote></details>

I am going to leave it as it is for now. Until we figure out what it means :-). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Comment by marcinslusarz** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[doc/libpmemfile-core.3.md, line 58 at r1](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27#-KermCzdSuPDH4KzI2OB:-KfB72vgZI8O3tbNJQwq:b-7r1ac7) ([raw file](https://github.com/marcinslusarz/pmemfile/blob/8732fe4214e0490a2d14e0490c8f437905983522/doc/libpmemfile-core.3.md#L58)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

Actually, we are moving data to and from DRAM.  Even if we read from the kernel fs, it's read into a dram buffer, than we write that data to pmemfile, we write from a dram buffer.  When we use NVML we have in place persistence. With pmemfile we have a buffer in between. Which is what I am trying to say.  Maybe I will just pull this out since it's confusing. And, honestly, isn't important to a user. 
</blockquote></details>

Ha. Now I understand what you ment :).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/pmemfile/27)*

**Pull request merged**